make/tests

```diff
@@ -1,32 +1,48 @@
 ##
-# LIBGTEST is the google test library
-# GTEST_MAIN is the file that contains the google test
+# Build test executables
 ##
-LIBGTEST = test/libgtest.a
-GTEST_MAIN = $(GTEST)/src/gtest_main.cc
-GTEST_CXXFLAGS += -isystem $(GTEST)/include -isystem $(GTEST)
 
-##
-# Build the google test library.
-$(LIBGTEST): $(LIBGTEST)(test/gtest.o)
-
-test/gtest.o: $(GTEST)/src/gtest-all.cc
-	@mkdir -p test
-	$(COMPILE.cc) -O$O $(GTEST_CXXFLAGS) $< $(OUTPUT_OPTION)
+test/%$(EXE) : CXXFLAGS += $(CXXFLAGS_GTEST)
+test/%$(EXE) : CPPFLAGS += $(CPPFLAGS_GTEST)
+test/%$(EXE) : INC_FIRST = -I $(if $(STAN),$(STAN)/src,src) -I $(if $(STAN),$(STAN),.)
+test/%$(EXE) : INC += $(INC_GTEST)
```

Comment:
I'm trying to use these for the first time and getting pretty turned around... Where is `INC` used, if anywhere?

I'm trying to figure out how to compile something that includes the correct Stan includes.